sdk/identity/azure-identity/src/client_certificate_credential.cpp

```diff
@@ -77,6 +77,38 @@ template <typename T> std::vector<uint8_t> ToUInt8Vector(T const& in)
   return outVec;
 }
 
+static std::string FindPemCertitificateContent(std::string const& path)
+{
+  auto pemContent{FileBodyStream(path).ReadToEnd()};
```

Comment:
`ReadPemCertificate()` already does open the file and read its contents. Can you make it so that x5c gets read as part of that process?

Reply:
Yeah, I wanted to do that, to reduce some of the duplication, but then the ifdefs unravel a bit. I plan to restructure this a bit more in a subsequent PR to make it cleaner (and attempt to reduce as much platform-specific code).

Right now, I prioritized keeping the logic as platform agnostic as possible.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/5990